aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/UserPoolConfiguration.kt

```diff
@@ -96,7 +97,7 @@ data class UserPoolConfiguration internal constructor(val builder: Builder) {
                     "https://$endpoint"
                 }
             } catch (e: Exception) {
-                throw AuthException(
+                throw ValidationException(
```

Comment:
I think this should be an Exception type, since UserPoolConfig is in codegen and validation exception is core package.

Reply:
If the issue is generic enough, throwing a core exception should be ok. In this case though, it may be more of a configuration exception than a validation exception. Thoughts?

Reply:
It will still be a problem as Configuration Exception is in auth package and this class is in codegen.